src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -296,4 +311,71 @@ public void deleteIsland(@NotNull Island island) {
         });
     }
 
+    /**
+     * Recalculates an island value
+     *
+     * @param island The specified Island
+     */
+    public void recalculateIsland(@NotNull Island island) {
+        //Reset their value
+        IridiumSkyblock.getInstance().getBlockValues().blockValues.keySet().stream().map(material -> IridiumSkyblock.getInstance().getIslandManager().getIslandBlock(island, material)).forEach(islandBlocks -> islandBlocks.ifPresent(blocks -> blocks.setAmount(0)));
```

Comment:
Why not `IridiumSkyblock.getInstance().getBlockValues().blockValues`? Just as an idea, I don't know why you'd do it in that way

Reply:
Not quite sure what you mean?

Reply:
`IridiumSkyblock.getInstance().getBlockValues().blockValues.clear()`, my bad

Reply:
Because thats the config file? Thats setting the block amounts to 0 for the island

Reply:
Ah, alright